holoviews/plotting/plot.py

```diff
@@ -687,9 +689,34 @@ def _get_norm_opts(self, obj):
                           if not any(spec[:i] in norm_opts.keys() for i in range(1, 4))})
         return norm_opts
 
+    @classmethod
+    def _merge_group_ranges(cls, ranges):
```

Comment:
As a classmethod that doesn't even use `cls`, it should be easy to set up some unit tests for this function (could even be a utility tbh).